static/js/fenced_code.js

```diff
@@ -52,6 +58,22 @@ function wrap_tex(tex) {
     }
 }
 
+function wrap_zpoiler(header, text) {
+    let output = []
+    const header_div_open_html = '<div class="zpoiler-block"><div class="zpoiler-header">'
+    const toggle_button_html = '<a class="zpoiler-button" aria-expanded="false"><span class="zpoiler-arrow zpoiler-button-closed"></span></a>'
+    const header_text = header
+    const end_header_start_content_html = '</div><div class="zpoiler-content zpoiler-content-closed" aria-hidden="true">'
+    const footer_html = '</div></div>'
```

Comment:
It's possible that this would read better as a handlebars template; feel free to convert it to that.  

A few notes on the HTML here, which will be an important and hard-to-change part of the Zulip API, since messages will be stored in the database indefinitely once rendered with a given HTML schema.
* I'd like to avoid having UI details like the `zpoiler-content-closed` classes present in the base HTML; we should omit them here.  It's possible that adjustments to the CSS to assume a `spoiler_content` is closed unless specified otherwise is the cleanest solution here (I think it would save code to do so, and just have a class for open spoilers); if not, we can add them before inserting into the DOM in `static/js/rendered_markdown.js`.  
* I want it to be easy to change the precise HTML for the toggle UI as well, which we can definitely insert via `rendered_markdown.js`.

I think that means the HTML would be something like this:
```
<div class="spoiler-block">
    <div class="spoiler-header">{{header}}</div>
     <div class="spoiler-content" aria-hidden="true">{{content}}</div>
</div>
```

@andersk do you have any further thoughts on how this should be represented as HTML?  I could imagine deciding to leave the `aria-hidden` detail to clients as well.

Reply:
We handled most of this. I don't think we can do a handlebars template here while also handling the concern around ensuring links/emoji in headers show up properly; the result of the template (markdown and all) would need to be passed to the client side stash function. We've instead cleaned up this function so the header and content don't need to be escaped, which I think should be more readable.

Reply:
OK, sounds good.  With the simpler HTML, this may be fine anyway.